backend/src/main/scala/cromwell/backend/package.scala

```diff
@@ -0,0 +1,31 @@
+package cromwell
+
+import com.typesafe.config.Config
+import cromwell.core.{WorkflowId, WorkflowOptions}
+import wdl4s.{CallInputs, NamespaceWithWorkflow, Task, WdlExpression}
+
+package object backend {
+
+  /**
+    * For uniquely identifying a job which has been or will be sent to the backend.
+    */
+  case class BackendJobDescriptorKey (callFqn: String, index: Option[Int], attempt: Int)
+
+  /**
+    * For passing to a BackendWorkflowActor at job execution, recovery or aborting time
+    */
+  case class BackendJobDescriptor(key: BackendJobDescriptorKey, task: Task, locallyQualifiedInputs: CallInputs)
```

Comment:
Any way that the second parameter can be a `Call` instead?  This will be especially useful if we ever use some of my proposed changes for wdl4s.  For example, the appropriate lookup function for instantiating commands and evaluating outputs would be a method on the `Call`.